Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -289,16 +289,22 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
                         } else {
                             auto lepton_weight = eventWeights_HH->GetProviderT<mc_corrections::LeptonWeights>(
                                     mc_corrections::WeightType::LeptonTrigIdIso);
-                            double total_lepton_weight = lepton_weight->Get(*event);
+
+                            double lepton_id_iso = lepton_weight->GetIdIsoWeight(*event,
+                                                         signalObjectSelector.GetTauVSeDiscriminator(channelId).second,
+                                                         signalObjectSelector.GetTauVSmuDiscriminator(channelId).second,
+                                                         signalObjectSelector.GetTauVSjetDiscriminator().second,
+                                                         unc_source, unc_scale);
+                            double lepton_trigger = lepton_weight->GetTriggerWeight(*event);
 
                             auto btag_weight = eventWeights_HH->GetProviderT<mc_corrections::BTagWeight>(
                                     mc_corrections::WeightType::BTag);
-                            double total_btag_weight = btag_weight->Get(*event);
+                            // double total_btag_weight = btag_weight->Get(*event);
 
                             double cross_section = (*summary)->cross_section > 0 ? (*summary)->cross_section :
                                                                                     sample.cross_section;
-                            const double weight = (*event)->weight_total * cross_section
-                                * ana_setup.int_lumi * total_lepton_weight * total_btag_weight
+                            const double weight = (*event)->weight_total * cross_section * (*event)->l1_prefiring_weight
```

Comment:
- apply b tag weight only if the current category has btag requirement
- apply L1 prefiring weight only for 2016 and 2017.